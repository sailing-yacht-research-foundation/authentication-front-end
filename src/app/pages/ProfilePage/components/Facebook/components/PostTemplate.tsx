import { selectUser } from 'app/pages/LoginPage/slice/selectors';
import React from 'react';
import { useSelector } from 'react-redux';
import { getProfilePicture } from 'utils/user-utils';
import { Post, PostUserWrapper, PostUserName, UserAvatar, PostCaption } from '../../LinkToProviders';

const PostTemplate = (props) => {

    const { index, post } = props;

    const user = useSelector(selectUser);

    const renderCaption = () => {
        if (post.attachments && post.attachments?.data?.length > 0) {
            if (post.attachments.data[0].description)
                return post.attachments.data[0].description;
            else if (post.attachments.data[0].title)
                return post.attachments.data[0].title;
        }

        return '';
    }

    return (
        <Post key={index}>
            <PostUserWrapper>
                <UserAvatar style={{ background: `url(${getProfilePicture(user)})`, backgroundSize: 'cover' }}></UserAvatar>
                <PostUserName>You</PostUserName>
            </PostUserWrapper>
            <PostCaption>{ renderCaption() }</PostCaption>
        </Post>
    )
}

export default PostTemplate