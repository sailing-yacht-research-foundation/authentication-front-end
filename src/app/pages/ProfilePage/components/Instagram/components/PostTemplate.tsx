import React from 'react';
import { Image } from 'antd';
import { getProfilePicture } from 'utils/user-utils';
import { useSelector } from 'react-redux';
import { selectUser } from 'app/pages/LoginPage/slice/selectors';
import { Post, PostUserWrapper, PostUserName, UserAvatar, PostCaption } from '../../LinkToProviders';

const PostTemplate = (props) => {
    const { post, index } = props;

    const user = useSelector(selectUser);

    return (
        <Post key={index}>
            <PostUserWrapper>
                <UserAvatar style={{ background: `url(${getProfilePicture(user)})`, backgroundSize: 'cover' }}></UserAvatar>
                <PostUserName>You</PostUserName>
            </PostUserWrapper>
            <PostCaption>{post.caption}</PostCaption>
            <Image src={post.media_url} />
        </Post>
    )
}

export default PostTemplate