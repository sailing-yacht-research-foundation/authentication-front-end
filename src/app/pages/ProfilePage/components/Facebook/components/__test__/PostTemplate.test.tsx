import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import PostTemplate from '../PostTemplate';
import MyProvider from 'app/components/Provider';

const shallowRenderer = createRenderer();

describe('<PostTemplate />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(
      <MyProvider>
        <PostTemplate post={null} />
      </MyProvider>);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
