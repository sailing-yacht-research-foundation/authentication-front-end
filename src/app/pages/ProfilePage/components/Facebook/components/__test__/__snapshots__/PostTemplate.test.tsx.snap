// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTemplate /> should render and match the snapshot 1`] = `
<Provider
  store={
    Object {
      "createReducer": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <PostTemplate
    post={null}
  />
</Provider>
`;
